src/Sparrow/Json/LazyStringParser.cs

```diff
@@ -231,17 +234,183 @@ public static bool TryParseTimeSpan(byte* buffer, int len, out TimeSpan ts)
+                goto Failed;
+
+            if (TryParseNumber4(buffer, 12, out var control) == false)
+                goto Failed;
+            else if (control is not 0)
+                goto Failed;
+
+
+            timeOnly = new TimeOnly(hours, minutes, seconds, milliseconds);
+            return true;
+
+            Failed:
+            timeOnly = default;
+            return false;
+        }
+
+        public static bool TryParseDateOnly(byte* buffer, int len, out DateOnly dateOnly)
+        {
+            if (len is not 10 || buffer[4] is not (byte)'-' || buffer[7] is not (byte)'-')
+                goto Failed;
+
+            if (TryParseNumber4(buffer, 0, out int year) == false)
+                goto Failed;
+
+            if (TryParseNumber2(buffer, 5, out int month) == false)
+                goto Failed;
+
+            if (TryParseNumber2(buffer, 8, out int day) == false)
+                goto Failed;
+
+            dateOnly = new DateOnly(year, month, day);
+            return true;
+
+            Failed:
+            dateOnly = default;
+            return false;
+        }
+
+        public static bool TryParseTimeOnly(byte* buffer, int len, out TimeOnly timeOnly)
+        {
+            if (len is not (8 or 16) || buffer[2] is not (byte)':' || buffer[5] is not (byte)':')
+                goto Failed;
+
+            if (TryParseNumber2(buffer, 0, out int hours) == false)
+                goto Failed;
+
+            if (TryParseNumber2(buffer, 3, out int minutes) == false)
+                goto Failed;
+
+            if (TryParseNumber2(buffer, 6, out int seconds) == false)
+                goto Failed;
+
+            int milliseconds = 0;
+            if (len is 16)
+            {
+                if (buffer[8] is not (byte)'.')
+                    goto Failed;
+                
+                    
+                if (TryParseNumber3(buffer, 9, out milliseconds) == false)
+                    goto Failed;
+
+                if (TryParseNumber4(buffer, 12, out var control) == false)
+                    goto Failed;
+                
+                if (control is not 0)
+                    goto Failed;
+            }
+
+            timeOnly = new TimeOnly(hours, minutes, seconds, milliseconds);
+            return true;
+
+            Failed:
+            timeOnly = default;
+            return false;
+        }
+#endif
+        
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Result TryParseTimeForQuery(char* buffer, int len, out DateTime dt, out DateTimeOffset dto,
+#if FEATURE_DATEONLY_TIMEONLY_SUPPORT
+            out DateOnly @do,
+            out TimeOnly to,
+#endif
+            bool properlyParseThreeDigitsMilliseconds)
+        {
+            dt = default;
+            dto = default;
+#if FEATURE_DATEONLY_TIMEONLY_SUPPORT
+            @do = default;
+            to = default;
+
+            if (TryParseDateOnly(buffer, len, out @do))
```

Comment:
In here, move the length check to this method, so we won't have method call overhead to know that this is a miss